compiler-collection.js

```diff
@@ -7,25 +7,33 @@ module.exports = (expireAfterSeconds) => {
     if (!compiler || !expireMs) {
        return;
     }
-    clearTimeout(compiler.expiredTimeout);
```

Comment:
I don't like the way this is done...  the timer is attached to the compiler, but it doesn't kill *that*  compiler, it just kills the compiler currently store for the repo. If we just had the timeout kill the compiler its attached to, there'd be less to worry about.

Reply:
Fixed!